### PR TITLE
Fix broken Jetpack e2e tests

### PR DESCRIPTION
**Pull request by rcanepa**
#### Changes proposed in this Pull Request

* Fix Jetpack e2e tests that the new release of the Pricing page broke. Specifically, this PR will fix tests from the `specs-jetpack-calypso/wp-jetpack-connect-spec.js`, `specs-jetpack-calypso/wp-jetpack-plans-spec.js`,  and `specs/wp-plan-purchase-spec.js` files.

#### Testing instructions

* Code review.
* Make sure tests pass.

Fixes 1164141197617539-as-1199221059177321

### CODE REVIEW

**Comment by mattgawarecki** [comment]
The code looks good to me, and GitHub's saying all the e2e tests passed except for `ci/wp-e2e-tests-full-jetpack`. I checked the test run there and it seems like those might be unrelated?

**Comment by rcanepa** [comment]
> The code looks good to me, and GitHub's saying all the e2e tests passed except for `ci/wp-e2e-tests-full-jetpack`. I checked the test run there and it seems like those might be unrelated?

Yes, those are unrelated to the changes we introduced with the new version of the plans grid. I think those should be fixed in a different PR.

**Comment by rcanepa** [comment]
@brbrr I noticed that a lot of tests are failing with the same error that says:

```
TimeoutError: Timed out waiting for element with css selector of '.site-selector .site a[aria-label*="calypsojetpackbleedingedge.mystagingwebsite.com"]' to be clickable
```

When I ran the suite locally, I noticed that those tests are failing because the Site Selector doesn't seem to have any site in it. Or the request that fetches them doesn't ever return.

image

Have you seen this issue before?

File: test/e2e/lib/pages/my-plan-page.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ export default class MyPlanPage extends AsyncBaseContainer {
 	async isSecurityPlan() {
 		return await driverHelper.isEventuallyPresentAndDisplayed(
 			this.driver,
-			By.css( '[data-e2e-product-slug="jetpack_security_daily"]' )
+			By.css( '.is-jetpack-security-v2' )
```
Comment (monsieur-z) [inline test/e2e/lib/pages/my-plan-page.js]: Could this break with new iterations of the pricing page, @rcanepa?

Comment (rcanepa) [reply test/e2e/lib/pages/my-plan-page.js]: Yeah. I'm going to improve it by using a data attribute. Currently, the HTML element doesn't have the data attribute so `'[data-e2e-product-slug="jetpack_security_daily"]'` doesn't find the element.

**Review by monsieur-z — APPROVED** [review]
LGTM!

**Comment by rcanepa** [comment]
I'll merge this and address the rest of the broken Jetpack tests in a follow-up PR.

**Comment by brbrr** [comment]
> Have you seen this issue before?

Sorry for the late response :/ 

Nope, I haven't. It might be related to the amount of sites for that user. They should be cleaned up daily, but I wonder if the clean up job still works

**Comment by brbrr** [comment]
turns out the clean up job wasn't running :facepalm: working on a fix

**Pull request merged**